src/features/users/users.repository.ts

```diff
@@ -0,0 +1,112 @@
+import type { Request } from 'express'
+import { prisma } from '../../adapters/prisma/client'
+
+export const transferOwnershipToUser = async ({
+  userId,
+  email,
+}: NonNullable<Request['user']>) => {
+  const usersToMigrate = await prisma.user.findMany({
+    where: {
+      id: {
+        not: userId,
+      },
+      email,
+    },
+    orderBy: {
+      createdAt: 'desc',
+    },
+    select: {
+      id: true,
+      name: true,
+      email: true,
+      createdAt: true,
+      updatedAt: true,
+    },
+  })
+
+  const [existingUser] = usersToMigrate
+
+  if (!existingUser) {
+    return
+  }
+
+  await prisma.user.upsert({
+    where: {
+      id: userId,
+    },
+    create: {
+      ...existingUser,
+      id: userId,
+    },
+    update: {
+      ...existingUser,
+      id: userId,
+    },
+  })
+
+  await Promise.all(
+    usersToMigrate.map(({ id }) =>
+      Promise.all([
+        prisma.groupAdministrator.updateMany({
+          where: {
+            userId: id,
+          },
+          data: {
+            userId,
+          },
+        }),
+        prisma.groupParticipant.updateMany({
+          where: {
+            userId: id,
+          },
+          data: {
+            userId,
+          },
+        }),
+        prisma.simulation.updateMany({
+          where: {
+            userId: id,
+          },
+          data: {
+            userId,
+          },
+        }),
+      ])
+    )
+  )
+
+  const oldPollsSimulations = await prisma.simulationPoll.findMany({
+    skip: 1,
```

Comment:
Why do we skip 1 here ?

Reply:
Because we want to keep the last one

Basically a user creates a simulation into a poll on device 1

In database we have one simulationPoll entry

Then the same user creates an other simulation into the same poll on device 2

In database we have two simulationPoll entries because we don't know this is the same user

Lucky for us, he/she left his/her email on device 1. When he/she logs in or do whatever action giving us the email we keep only the last simulation done for poll (here the one on device 2)